.github/workflows/build_dev_documentation.yml

```diff
@@ -0,0 +1,108 @@
+name: Build dev documentation
+
+on:
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  build_and_package:
+    runs-on: ubuntu-latest
+    container:
+      image: huggingface/transformers-doc-builder # Not sure if we can also use this image for optimum
+    env:
+      COMMIT_SHA: ${{ github.event.pull_request.head.sha }}
+      PR_NUMBER: ${{ github.event.number }}
+      EVENT_CONTEXT: ${{ toJSON(github.event) }}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: 'huggingface/doc-builder'
+          path: doc-builder
+
+      - uses: actions/checkout@v2
+        with:
+          repository: 'huggingface/optimum'
+          path: optimum
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+
+      - name: Set env
+        run: |
+          echo "WRITE=$(echo 'ghp_'$(wget -qO- lysand.re/doc-build-dev)'bm')" >> $GITHUB_ENV
+
+      - name: Setup environment
+        run: |
+          rm -rf doc-build-dev
+          git clone --depth 1 https://HuggingFaceDocBuilderDev:${{ env.WRITE }}@github.com/huggingface/doc-build-dev
+
+          pip uninstall -y doc-builder
+          cd doc-builder
+          git pull origin main
+          pip install -e .
+          cd ..
+
+          cd optimum
+          pip install .[dev]
+          cd ..
+
+      - name: Setup git
+        run: |
+          git config --global user.name "Hugging Face Doc Builder"
+          git config --global user.email docs@huggingface.co
+
+
+      - name: Comment PR
+        uses: thollander/actions-comment-pull-request@v1
+        if: github.event.action == 'opened'
+
+        with:
+          message: 'The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/optimum/pr_${{ env.PR_NUMBER }}). All of your documentation changes will be reflected on that endpoint.'
```

Comment:
@Pierrci this is the step I'm unsure of - how does one actually get the docs to render at `moon-ci-doc.huggingface.co`? Right now I'm getting a 404, which is expected I guess :)

Reply:
`moon-landing` needs to be adapted to support optimum - you should probably open a PR on top of https://github.com/huggingface/moon-landing/pull/2291 (as it's currently what's deployed in moon-ci-docs) that adds the exact same changes but for optimum, and we can deploy it so that both accelerate and optimum are supported cc @mishig25

Reply:
https://huggingface.slack.com/archives/C02GLJ5S0E9/p1646815321086879?thread_ts=1646410559.029439&cid=C02GLJ5S0E9

Reply:
Hey guys, should I wait for https://github.com/huggingface/moon-landing/pull/2291 to be merged before merging this PR?

Reply:
Nope, it would rather be the opposite actually :)

Reply:
OK, I'm gonna merge this and iterate if needed :)